amd/src/app/block/element.js

```diff
@@ -170,6 +170,70 @@ export default class BlockElement {
         const bulkDeleteButton = this.#element.querySelector('#block_sharing_cart_bulk_delete_confirm');
 
         const checkboxSelector = '.sharing_cart_item input[data-action="bulk_select"][type="checkbox"]';
+        const selectAllCheckbox = this.#element.querySelector('#select_all_box');
+        const selectAllLabel = this.#element.querySelector('#select_all_label');
+
+        const cancelBulkDeleteButton = this.#element.querySelector('#block_sharing_cart_cancel_bulk_delete');
+        const bulkDeleteTrigger = this.#element.querySelector('#block_sharing_cart_bulk_delete');
+
+
+
+        const getItemCheckboxes = () => this.#element.querySelectorAll(checkboxSelector);
+
+        const updateBulkDeleteButtonState = () => {
+            bulkDeleteButton.disabled = !Array.from(getItemCheckboxes()).some(checkbox => checkbox.checked);
+        };
```

Comment:
Duplicate code, can just use `this.updateBulkDeleteButtonState()` which you've also created